rest-api/dao/participant_summary_dao.py

```diff
@@ -336,6 +395,62 @@ def calculate_enrollment_status(self, consent_for_study_enrollment_and_ehr,
       return EnrollmentStatus.MEMBER
     return EnrollmentStatus.INTERESTED
 
+  def calculate_member_time(self, consent_for_study_enrollment_and_ehr, participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.enrollmentStatusMemberTime is not None:
+      return participant_summary.enrollmentStatusMemberTime
+    elif consent_for_study_enrollment_and_ehr:
+      return participant_summary.consentForElectronicHealthRecordsTime
+    else:
+      return None
+
+  def calculate_core_stored_sample_time(self, consent_for_study_enrollment_and_ehr,
+                                        participant_summary):
+    if consent_for_study_enrollment_and_ehr and \
+      participant_summary.numCompletedBaselinePPIModules == \
+      self._get_num_baseline_ppi_modules() and \
+      participant_summary.physicalMeasurementsStatus == PhysicalMeasurementsStatus.COMPLETED and \
+      participant_summary.samplesToIsolateDNA == SampleStatus.RECEIVED:
+
+      if participant_summary.enrollmentStatusCoreStoredSampleTime:
```

Comment:
add this logic to "stored" but not for "ordered", because when a biobank order cancel request comes, we need to re-calculate the order sample timestamp.

Reply:
So even in that circumstance, I think we want this kind of logic... what you could do is calculate the the max core sample time, and if it's not None, then use the existing value if there is one, otherwise use the new value.